### PR TITLE
chore: publish binaries

### PR DESCRIPTION
**Pull request by morenol**
Publish binary for DEV releases and also for tagged releases

### CODE REVIEW

**Review by peterhuene — COMMENTED** [review]
Looks great, just some minor comments. Thanks for putting this together!

File: .github/workflows/publish-binaries.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main' && github.repository == 'bytecodealliance/cargo-component'
+    name: Create dev release
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Login GH CLI
+        run: gh auth login --with-token < <(echo ${{ secrets.GITHUB_TOKEN }})
+      - name: Delete old dev release
+        run: gh release delete -R infinyon/cargo-component dev -y || true
```
[the remark this answers is not in the document]
Comment (morenol) [reply .github/workflows/publish-binaries.yml:20]: right, I just was using that to test. I am sorry

[the remark this answers is not in the document]
Comment (peterhuene) [reply .github/workflows/publish-binaries.yml:20]: No reason to apologize; I assumed that was the reason 😄

File: .github/workflows/publish-binaries.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      - name: Create new latest release
+        run: gh release create -R infinyon/cargo-component dev --notes "Published artifacts from the latest build"
+
+  publish_dev_release:
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main' && github.repository == 'bytecodealliance/cargo-component'
+    name: Publish to dev release
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
```
[the remark this answers is not in the document]
Comment (morenol) [reply .github/workflows/publish-binaries.yml:29]: Added in https://github.com/bytecodealliance/cargo-component/pull/161/commits/000a655aca22e95a18f7cd16cf7af4892eb85f6c

**Review by peterhuene — APPROVED** [review]
LGTM. Thanks for this!

**Pull request merged**